### DISCUSSION TITLE
How to use "User build Problems view" for a licensed IBM Dependency Based Build customer

### DISCUSSION THREAD

**Discussion by FALLAI-Denis** [IBM/zopeneditor-about#414] [Q&A]
## Description of the enhancement requested

Hi,

We are a customer who has a license for the IBM Dependency Based Build product, "standalone" mode. We do not have IBM Developer for Z licenses.

We want to use the [User build Problems view](https://www.ibm.com/docs/en/developer-for-zos/16.0?topic=capabilities-user-build-problems-view) feature which is related to IBM Dependency Based Build and not to IBM Developer for Z, (for us it does not make sense to acquire IDz licenses when we have chosen VS Code and IBM Z Open Editor, possibly we could consider acquiring Wazi Developer licenses provided that they do not impose the acquisition of IDz licenses).
Moreover, the dependency is more with zAppbuild, which is open source and free of charge, than with IBM DBB which only manages the build results metadata repository, and is a licensed product.

![image](https://github.com/IBM/zopeneditor-about/assets/63861965/1b18b64d-1b11-48f0-aba1-4480aedda715)

More precisely, we do not use the IBM User Build functionality because to date it does not deal with all the types of components that we need, and it does not manage the deployment which must follow after a user Build, (and the deployment procedures cannot be managed by IBM DBB itself because they depend on many contextual aspects outside the scope of IBM DBB, which is above all a CI tool and not a CD tool, moreover we are considering the implementation of Wazi Deploy in consistency with DBB).

We have developed our own build procedure, triggered from VS Code, and which retrieves the build result information, (made by IBM DBB "userBuild"), including the compilation listing, (COBOL for the use case).

A priori the routine for analyzing the compilation listing to extract errors must exist internally in the IBM User Build procedures, it would therefore be "sufficient" to make them accessible via an exposed API...

One element that intrigues me is how the functionality manages to make the link between line numbers in a compilation listing, which are impacted by copybook inclusions, but also conditional compilation, (our recurring topic), *CONTROL NOSOURCE directives, etc., and line numbers in the original source code... unless the Z Open Editor COBOL language server is capable of internally constructing an expanded source code identical to that of the IBM Enterprise COBOL compiler.

Thanks.

**Comment by phaumer** [comment]
@FALLAI-Denis I am moving this item to the Discussions section as that might be a better place for it. To implement the problems view we do parse a log file provided by DBB and the Compiler to be able to generate the diagnostics entries in the problems view. The details could perhaps be discussed and provided via tech support request to the DBB team.

**Comment by FALLAI-Denis** [comment]
Hi,

This needs to acquire licences for advanced functions of IBM Z Open Editor, and use a standard zAppbuild installation integrated with IDzEE server, (use of ADEXIT(ELAXGMUX) provided by IDzEE server...).
